src/MoonSharp.Interpreter/CoreLib/OsTimeModule.cs

```diff
@@ -123,7 +123,9 @@ public static DynValue date(ScriptExecutionContext executionContext, CallbackArg
 
 				try
 				{
-					reference = TimeZoneInfo.ConvertTimeFromUtc(reference, TimeZoneInfo.Local);
+					TimeZoneInfo localTimeZoneInfo = executionContext.OwnerScript.Options.LocalTimeZoneInfo ?? TimeZoneInfo.Local;
```

Comment:
Logic needs to be added to:

[src/MoonSharp.Interpreter/_Projects/MoonSharp.Interpreter.netcore/src/CoreLib/OsTimeModule.cs ](src/MoonSharp.Interpreter/_Projects/MoonSharp.Interpreter.netcore/src/CoreLib/OsTimeModule.cs)
[src/Unity/MoonSharp/Assets/Plugins/MoonSharp/Interpreter/CoreLib/OsTimeModule.cs](src/Unity/MoonSharp/Assets/Plugins/MoonSharp/Interpreter/CoreLib/OsTimeModule.cs)

If adding this in to the current system; as we push to clean this up it will change, but for now this PR will be accepted if the additional locations are updated.